components/raftstore-v2/src/operation/command/admin/merge/commit.rs

```diff
@@ -314,19 +308,16 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
                 region
             );
             // Best effort. Remove when trim check is implemented.
-            if self.storage().has_dirty_data() {
-                info!(self.logger, "ignore commit merge because of dirty data");
-                None
-            } else {
-                Some(true)
+            if !self.storage().has_dirty_data() && self.is_leader() {
```

Comment:
As said in the above comment, this check can be removed.

Reply:
Which comment? This line is not related to rollback.

Reply:
> // Best effort. Remove when trim check is implemented.